org.eclipse.scout.rt.jackson.test/src/test/java/org/eclipse/scout/rt/jackson/dataobject/DoEntitySerializerAttributeNameComparatorTest.java

```diff
@@ -42,4 +43,13 @@ public void testCompare() {
     assertTrue(comparator.compare(context.getTypeVersionAttributeName(), "alfa") < 0);
     assertTrue(comparator.compare("alfa", context.getContributionsAttributeName()) < 0);
   }
+
+  @Test
+  public void testInitMayBeCalledOnlyOnce() {
+    ScoutDataObjectModuleContext context1 = BEANS.get(ScoutDataObjectModule.class).getModuleContext();
+    ScoutDataObjectModuleContext context2 = BEANS.get(ScoutDataObjectModule.class).getModuleContext();
+    DoEntitySerializerAttributeNameComparator comparator = BEANS.get(DoEntitySerializerAttributeNameComparator.class)
+        .init(context1);
+    assertThrows(AssertionException.class, () -> comparator.init(context2));
+  }
 }
```

Comment:
An additional test would be nice, demonstrating that two comparators (differently configured) may use their own sorting.